src/java.base/unix/classes/sun/nio/fs/UnixFileSystem.java

```diff
@@ -1011,6 +1000,19 @@ void copy(final UnixPath source,
         try {
             sourceAttrs = UnixFileAttributes.get(source, flags.followLinks);
         } catch (UnixException x) {
+            if (x.errno() == UnixConstants.ENOENT) {
+                if (Files.isSymbolicLink(source)) {
+                    Path symbolicLink = null;
+                    try {
+                        symbolicLink = Files.readSymbolicLink(source);
+                    } catch (IOException e) {
+                        //use rethrowAsIOException
+                    }
+                    if (symbolicLink != null) {
+                        throw new NoSuchFileException(source.toString(), null, "due to " + symbolicLink);
+                    }
```

Comment:
The default provider implementation can't use Files.* methods, otherwise you end up with recursive usage.

In any case, I think the starting out to decide if anything should change as the existing NoSuchFileException is not wrong.

Reply:
Thanks for review
@AlanBateman 
The NoSuchFileException in this case is correct,The code is to add contextual information for NoSuchFileException.


In this  case, the symlinked file does not exist, the symlink exist.

NoSuchFileException info is symlink.

`ls path to symlink`   The detection file exists.This is supposed to detect the  symlinked file.Expect to add symbolic link information in the NoSuchFileException

before
```
Exception in thread "main" java.nio.file.NoSuchFileException: /data/testfiles/testlink
```
after
```
Exception in thread "main" java.nio.file.NoSuchFileException: /data/testfiles/testlink: due to testfile
```




The condition for throwing a NoSuchFileException has not changed

```
    private IOException translateToIOException(String file, String other) {
……
        if (errno() == UnixConstants.ENOENT)
            return new NoSuchFileException(file, other, null);
……
```

Reply:
> The default provider implementation can't use Files.* methods, otherwise you end up with recursive usage.

`UnixFileSystemProvider` is what I think ought to be used instead.

> In any case, I think the starting out to decide if anything should change as the existing NoSuchFileException is not wrong.

Please see [2. Socialize your change](https://openjdk.org/guide/#socialize-your-change).

Reply:
@AlanBateman @bplb 
sorry for late,Thanks for guiding the process